### PR TITLE
Add mechanism for comparison of setup.py and its equivalent setup.cfg

### LINKED ISSUE #1699: Add mechanism for side-by-side comparison of setup.py and its equivalent setup.cfg
We have many documentation examples that are purely declarative and are either documented as `setup.py` or `setup.cfg`. It would be really awesome if, for each of these, we had the option to have either both versions side-by-side or, even better, in a sort of "tabbed container", like the one in the [code sample at the bottom of this example](https://leetcode.com/articles/median-of-two-sorted-arrays/).

Requirements for this:

1. Cannot *link to* any third-party javascript dependencies. Ideally we wouldn't use any at all, but if you do they must be vendored in the documentation.
2. If javascript is disabled, it has to fall back to something intelligible.

Ideally it would be implemented in pure CSS / HTML if that's at all possible.

### PR DESCRIPTION
**Pull request by amy-lei**
## Summary of changes

Closes #1699

Add the [sphinx-inline-tabs](https://github.com/pradyunsg/sphinx-inline-tabs) extension to allow tabbed containers for comparing `setup.py` and `setup.cfg`. The tabs works even when JS is disabled. This PR also replaces instances of adjacent code-blocks with the new extension. 

However, I wasn't sure if there was an issue with the default styling. It currently looks like this: 
image
The only thing that stood out and possibly needed overriding was the accent color of the tabs.

### Pull Request Checklist
- [ ] ~~Changes have tests~~
- [x] News fragment added in [`changelog.d/`].
  _(See [documentation][PR docs] for details)_

[`changelog.d/`]: https://github.com/pypa/setuptools/tree/master/changelog.d
[PR docs]:
https://setuptools.readthedocs.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-extensions = ['sphinx.ext.autodoc', 'jaraco.packaging.sphinx', 'rst.linker']
+extensions = ['sphinx.ext.autodoc', 'jaraco.packaging.sphinx', 'rst.linker', 'sphinx_inline_tabs']
```
Comment (jaraco) [inline docs/conf.py:1]: This change is causing a nitpicky linter to fail the tests. I'd normally let you fix it, but I probably want to make some concession for other concerns (integration with the [upstream template](https://github.com/jaraco/skeleton)), so I can make the adjustments.

Comment (jaraco) [reply docs/conf.py]: I applied a fix in 265fb0eda.

**Review by jaraco — APPROVED** [review]
This is beautiful. I honestly did not expect this change to be possible, but the solution you found and implemented is simple and elegant. Thoroughly impressed and grateful.

**Comment by webknjaz** [comment]
Merging this PR has broken the RTD setup 3 weeks ago. Since then, all the builds are failing: https://github.com/pypa/setuptools/issues/2614.
I think we should enable the PR RTD builds toggle (https://github.com/pypa/setuptools/issues/2469) to notice such problems early.

**Pull request merged**